samples/tfm/provisioning_image/Kconfig.sysbuild

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menu "Network core configuration"
+	depends on SUPPORT_NETCORE
+
+choice NETCORE
+	prompt "Netcore image"
+	default NETCORE_PROVISIONING_IMAGE_NET_CORE if NETCORE_REMOTE_BOARD_TARGET_CPUCLUSTER != ""
+
+config NETCORE_PROVISIONING_IMAGE_NET_CORE
+	bool "Provisioning"
+	help
+	  Include tfm provisioning network core image.
+
+endchoice
+
+if !NETCORE_NONE
+
+config NETCORE_IMAGE_NAME
+	default "provisioning_image_net_core" if NETCORE_PROVISIONING_IMAGE_NET_CORE
+
+config NETCORE_IMAGE_PATH
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/tfm/provisioning_image_net_core" if NETCORE_PROVISIONING_IMAGE_NET_CORE
```

Comment:
provisioning_image_net_core sample was used for the singular purpose of locking the app_protect in network core, when CONFIG_TFM_NRF_PROVISIONING is used. This was demonstrated with tfm_psa_template sample.

tfm_psa_template sample was then changed so that network core image could be used and we could still guarantee that app_protect is locked. So the provisioning_image_net_core became unnecessary. However, it was left as the locking of the app_protect with tfm_psa_template required sysbuild.

I am not sure if there would still be some use for the provisioning_image_net_core sample, but I don't think that it should be used as default with provisioning sample. Doing so, breaks the tfm_psa_template sample, which expects to flash network core image.

Reply:
Discussed this with @frkv. The provisioning_image_net_core sample should be removed.

This would need the following changes (in the future documentation PR):
- Remove the reference to `provisioning_image_net_core` from nrf/samples/tfm/provisioning_image/README.rst
- Migration guide notice from removal, which would direct to use tfm_psa_template, which will take care of approtect locking.